### PR TITLE
chore(deps) bump lua-resty-http from 0.15 to 0.16.1

### PR DESCRIPTION
**Pull request by bungle**
### Summary

Upgrades `lua-resty-http` from 0.15 to 0.16.1 and removes the deprecated ways of using it.

### CODE REVIEW

File: spec/helpers.lua [hunk 1]
```diff
@@ -662,7 +662,51 @@ for _, method_name in ipairs({"get", "post", "put", "patch", "delete"}) do
...
 end
 
---- Creates a http client.
+
+--- Creates a http client from options.
+-- Instead of using this client, you'll probably want to use the pre-configured
+-- clients available as `proxy_client`, `admin_client`, etc. because these come
+-- pre-configured and connected to the underlying Kong test instance.
+--
+-- @function http_client
```
Comment (Tieske) [inline spec/helpers.lua:671]: name should be `http_client_opts`

File: spec/helpers.lua [hunk 2]
```diff
@@ -678,14 +722,15 @@ end
 -- @see admin_client
 -- @see admin_ssl_client
 local function http_client(host, port, timeout)
-  timeout = timeout or 10000
-  local self = setmetatable(assert(http.new()), resty_http_proxy_mt)
-  local _, err = self:connect(host, port)
-  if err then
-    error("Could not connect to " .. host .. ":" .. port .. ": " .. err)
+  if type(host) == "table" then
```
Comment (Tieske) [inline spec/helpers.lua:725]: should we add a deprecation warning/log in this function?

**Review by Tieske — DISMISSED** [review]
overal looks good. Some minor things.

Don't like so many `ssl_verify = false` though, we should fix those

**Review by pintsized — APPROVED** [review]
Nice! LGTM :+1:

**Comment by bungle** [comment]
> Don't like so many `ssl_verify = false` though, we should fix those

I'll create a backlog card for that.

**Pull request merged**